### PR TITLE
Assert that the required env vars exist

### PR DESCRIPTION
**Pull request by keirog**
Logs an error/triggers a Sentry error if any required environment
variables are missing, according to the app.json manifest.

Runs at:

- npm postinstall (so it’s caught by Heroku before releasing to prod)
- on app start-up (in case we make a mistake in the config vars UI)

When everything is missing during deployment 🙈:

Heroku build log showing an error that mentions every missing env var

When everything is missing at start up 🙈:
Terminal showing an error that mentions every missing env var on start-up

_This time_ it should actually fail to build, due to the [purposely-missing variable called `PURPOSELY_MISSING_VAR`](https://github.com/Financial-Times/email-send-api/blob/master/app.json#L44) 😄
(I misinterpreted some Heroku documentation before and thought the app.json would be enough)

 🐿 v2.10.2

### CODE REVIEW

**Comment by keirog** [comment]
The maintainer of that library sits in OSB 3A 😄

File: assert-env.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const assertHerokuEnv = require('@quarterto/assert-heroku-env');
```
Comment (AbdullahAli) [inline assert-env.js]: Are we able to copy the code directly into here.  I feel scared to rely on an external GitHubGist

File: assert-env.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+const logger = require('./logger');
+const sentry = require('./sentry').init();
+
+if (process.env.NODE_ENV === 'production') {
+	assertHerokuEnv(warnings => {
+		logger.error({ event: 'MISSING_ENV_VARS', warnings });
+		sentry.captureException(warnings, { level: 'warning' });
```
Comment (AbdullahAli) [inline assert-env.js]: Should we raise this as an `error` level instead of `warning`?

File: server.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 require('dotenv').load({ silent: true });
 
+
```
Comment (AbdullahAli) [inline server.js]: 🚓

File: server.js [hunk 4]
```diff
@@ -26,6 +26,8 @@ process.on('unhandledRejection', (rejectionReason, rejectedPromise) => {
 	});
 });
 
+require('./assert-env');
```
Comment (AbdullahAli) [inline server.js]: What effect does adding this here have? Is the `postinstall` script enough?

**Comment by AbdullahAli** [comment]
Do we need to remove `PURPOSELY_MISSING_VAR` to avoid the build breaking?

File: server.js (hunk 3 above)
Comment (keirog) [reply server.js]: 🙇‍♂️ How embarrassing

**Comment by keirog** [comment]
(As agreed IRL, I’ll do a test deploy of this branch first, then remove the dummy var)

**Pull request merged**